plugin/src/main/java/at/helpch/chatchat/listener/ChatListener.java

```diff
@@ -86,6 +80,11 @@ public void onChat(final AsyncPlayerChatEvent event) {
         // Cancel the event if the message doesn't end up being sent
         // This only happens if the message contains illegal characters or if the ChatChatEvent is canceled.
         event.setCancelled(!MessageProcessor.process(plugin, user, channel, message, event.isAsynchronous()));
+
+        // Ensure the user still has the channel permission, if not reset them back to the default channel
+        if (!channel.isUsableBy(user)) {
+            user.channel(channel);
+        }
```

Comment:
This will not be needed if the functionality will be changed according to the other comment in this review.